src/legacy/ui/public/courier/search_strategy/default_search_strategy.js

```diff
@@ -57,68 +58,86 @@ async function serializeAllFetchParams(fetchParams, searchRequests, serializeFet
+  // Flatten the searchSource within each searchRequest to get the fetch params,
+  // e.g. body, filters, index pattern, query.
+  const allFetchParams = await getAllFetchParams(searchRequests, Promise);
 
-    const searching = es.msearch(msearchParams);
+  // Serialize the fetch params into a format suitable for the body of an ES query.
+  const {
+    serializedFetchParams,
+    failedSearchRequests,
+  } = await serializeAllFetchParams(allFetchParams, searchRequests, serializeFetchParams);
 
+  if (serializedFetchParams.trim() === '') {
     return {
-      // Munge data into shape expected by consumer.
-      searching: new Promise((resolve, reject) => {
-        // Unwrap the responses object returned by the ES client.
-        searching.then(({ responses }) => {
-          resolve(responses);
-        }).catch(error => {
-          // Format ES client error as a SearchError.
-          const { statusCode, displayName, message, path } = error;
-
-          const searchError = new SearchError({
-            status: statusCode,
-            title: displayName,
-            message,
-            path,
-          });
-
-          reject(searchError);
-        });
-      }),
-      abort: searching.abort,
       failedSearchRequests,
     };
-  },
+  }
+  const msearchParams = {
+    ...getMSearchParams(config),
+    body: serializedFetchParams,
+  };
 
-  isViable: (indexPattern) => {
-    if (!indexPattern) {
-      return false;
-    }
+  const searching = es.msearch(msearchParams);
 
-    return isDefaultTypeIndexPattern(indexPattern);
-  },
-};
+  return {
+    // Munge data into shape expected by consumer.
+    searching: new Promise((resolve, reject) => {
+      // Unwrap the responses object returned by the ES client.
+      searching.then(({ responses }) => {
+        resolve(responses);
+      }).catch(error => {
+        // Format ES client error as a SearchError.
+        const { statusCode, displayName, message, path } = error;
+
+        const searchError = new SearchError({
+          status: statusCode,
+          title: displayName,
+          message,
+          path,
+        });
+
+        reject(searchError);
+      });
+    }),
+    abort: searching.abort,
+    failedSearchRequests,
+  };
+}
+
+function search({ searchRequests, es, Promise, config, sessionId, esShardTimeout }) {
+  const abortController = new AbortController();
+  const searchParams = getSearchParams(config, sessionId, esShardTimeout);
+  const promises = searchRequests.map(async searchRequest => {
+    const { index, body } = searchRequest.fetchParams = await searchRequest.getFetchParams();
+    try {
+      const promise = es.search({ index: index.title || index, body, ...searchParams });
+      abortController.signal.addEventListener('abort', promise.abort);
+      return promise;
+    } catch (e) {
+      return JSON.parse(e.response);
+    }
+  });
+  return {
+    searching: Promise.all(promises),
```

Comment:
Shall we abort this promise, too?

```js
const searching = Promise.all(promises);
abortController.signal.addEventListener('abort', searching.abort);
return {
  searching,
  // ...
};
```

Reply:
And then `searching` will get rejected anyways because one of the promises it awaits on will be aborted, right?

Reply:
I don't know if our build supports it, but there is [`Promise.allSettled`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled).